crates/module-system/module-implementations/sov-evm/src/call.rs

```
@@ -46,84 +48,36 @@ where
         // Inside the EVM, we use nonces only for the CREATE operation.
         // The uniqueness check was performed before the call was dispatched.
         let account_nonce = self.get_account_nonce(signer, state)?;
-
-        let evm_tx: TransactionSigned = convert_to_transaction_signed(message.rlp)?;
-
+        let gas_limit = state.gas_limit()?;
+        let tx_env = create_tx_env(&tx, signer, account_nonce, gas_limit.as_ref()[0]);
         let block_env = self
             .block_env
             .get(state)?
-            .expect("Pending block must be set");
+            .context("Pending block must be set")?;
+        let transaction = TransactionSignedAndRecovered {
+            signer,
+            signed_transaction: tx,
+            block_number: block_env.number.to::<u64>(),
+        };
 
-        let cfg = self.cfg(state)?.expect("Evm config must be set");
+        let cfg = self.cfg(state)?.context("Evm config must be set")?;
```

I think we should panic here instead of returning early. If cfg is not set, it indicates a serious issue we can't recover from, so panicking is appropriate.

---

I’ve updated the tracking issue. At some point, we’ll review all `.expect` calls in the EVM module and add comments explaining why they’re justified (or remove them)